src/paste3/paste.py

```diff
@@ -146,9 +140,13 @@ def pairwise_align(
     else:
         b = nx.from_numpy(b_distribution)
 
-    if isinstance(nx, ot.backend.TorchBackend) and use_gpu:
-        a = a.cuda()
-        b = b.cuda()
+    if isinstance(nx, ot.backend.TorchBackend):
```

Comment:
Grouped all conditionals for `TorchBackend` in one place. The `a = a.double()` is needed because of the difference between the returned dtypes between `ot.backend.NumpyBackend().ones(..)` (double-precision)) and `ot.backend.TorchBackend.ones(..)` (single-precision)